src/tools.rs

```diff
@@ -670,6 +670,16 @@ pub(crate) fn parse_receive_headers(headers: &Headers) -> String {
         .join("\n")
 }
 
+pub(crate) fn single_value<T>(collection: impl IntoIterator<Item = T>) -> Option<T> {
+    let mut iter = collection.into_iter();
+    if let Some(value) = iter.next() {
+        if iter.next().is_none() {
+            return Some(value);
+        }
+    }
+    None
```

Comment:
or
```
let ret = iter.next();
if iter.count() > 0 {
    return None;
}
ret

Reply:
`iter.count()` has to go through the whole iter, though. Which is no problem in most cases, but it's nicer to not have to worry about it.